xbmc/epg/EpgContainer.cpp

```diff
@@ -245,16 +245,17 @@ bool CEpgContainer::PersistAll(void)
 {
   bool bReturn(true);
   m_critSection.lock();
-  std::map<unsigned int, CEpg*> copy = m_epgs;
+  EPGMAP copy = m_epgs;
   m_critSection.unlock();
 
-  for (EPGMAP_CITR it = copy.begin(); it != copy.end() && !m_bStop; it++)
+  for (const auto &epgEntry : copy)
   {
-    CEpg *epg = it->second;
+    CEpg *epg = epgEntry.second;
     if (epg && epg->NeedsSave())
-    {
       bReturn &= epg->Persist();
-    }
+
+    if (m_bStop)
```

Comment:
you can't use `const auto` as it makes the iterator `it` const not the element itself. so `++` and `--` won't work.  You can only use `for (auto it = ..)` but you need to const cast the vector/map otherwise the elements are non-const.

Example: "non-const" elements
```
std::vector<int> test = {1, 2, 3, 4};
for (auto it = test.begin(); it != test.end(); ++it)
  (*it) = 1;
```
Example: "const" elements
```
const std::vector<int> test = {1, 2, 3, 4};
for (auto it = test.begin(); it != test.end(); ++it)
  (*it) = 1; // won't work as element is const
```